enhancements/user-monitoring.md

```diff
@@ -0,0 +1,248 @@
+The user monitoring Prometheus Operator will be deployed in the `openshift-user-monitoring` namespace.
+
+Currently, cluster monitoring operator scans all namespaces filtering openshift namespaces labelled with
+`openshift.io/cluster-monitoring: "true"` and takes this set of namespaces as the _allowed list_ of reconcilable
+namespaces for the existing cluster monitoring prometheus operator.
+
+The same set of openshift namespaces that exist today will be configured as a _deny list_ of reconcilable namespaces
+at the user monitoring prometheus operator.
+Deny listing of namespaces was added recently in prometheus operator, see [3].
+
+To make sure that users cannot add Prometheus instances in user namespaces,
+a new WIP feature [4] will ensure that reconcilation of Prometheus custom resources
+will be limited to the `openshift-user-monitoring` namespace for the user monitoring prometheus operator
+and `openshift-monitoring` namespace for the cluster monitoring prometheus operator.
+
+![](user-namespaces-custom-resources.png)
+
+User monitoring Prometheus Operator (green) then will reconcile Prometheus custom resources
+in the `openshift-user-monitoring` namespace only.
+It ignores Prometheus custom resources in any other namespace.
+The same holds true for the cluster monitoring prometheus operator (red).
+
+#### Aggregation
+
+The front facing Kubernetes service has to make it as simple as possible
+for integrators to leverage both user monitoring as well as cluster monitoring.
+
+This is achieved by using Thanos Querier which allows to aggregate
+and optionally deduplicate the cluster and user monitoring metrics backends
+under single Prometheus Query endpoint.
+
+Usage of the Thanos querier component inside cluster monitoring has many advantages:
+1. Run-time deduplication of metrics available today.
+2. Potentially gap-filling of metrics series in the Prometheus HA user and cluster monitoring setup in the future.
+3. Potentially add the possibility of adding a query cache for increased performance in the future.
+
+See [5] and [6] for more details.
+
+### Tenancy
+
+![](user-monitoring-request.png)
+
+Tenancy is achieved by leveraging the existing topology that is protecting cluster monitoring Prometheus already today.
+The kube-rbac-proxy sidecar will be deployed along with prom-label-proxy in the Thanos Querier Deployment.
+
+### Risks and Mitigations
+
+Risks are mainly in the area of security, namely in the realm of accessing metrics a user has no permissions for.
+This is achieved by reusing the existing tenancy stack as well as with the newly added features for allow-lists,
+deny-lists and reconcilation limiting settings for prometheus operator.
+
+## Design Details
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+N/A
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
```

Comment:
Was this meant as a link or a ref to something?

Reply:
this is from the enhancement template :man_shrugging: